javascript/ecmascript-2015/function-improvements/arrow-functions-part-2.md

```diff
@@ -40,26 +40,51 @@ links:
 ---
 ## Content
 
-Arrow Functions(`() => {}`) differ from anonymous keyword functions (`function(){}`) by not changing the value of `this`.
+Arrow Functions(`() => {}`) differ from anonymous keyword functions (`function(){}`) by not changing the value of `this`. Before **ES6**, every time you would call a function it would define its own `this` value, depending on how the function was called.
+
+Let's take a look at how this `Counter()` constructor is being defined: 
 
 ```js
 window.counter = 0;
 
 function Counter() {
+  // The Counter() constructor defines `this`
```

Comment:
So this isn't true. We can make `this` of the `Counter` be whatever we want because `this` is a runtime property, it's not statically defined when we write a function.

```js
// Counter() expects `this`
// to be an instance of itself
```